### PR TITLE
add: javascript snippet "voice.js" and accordingly update "edit_event…

### PR DESCRIPTION
**Pull request by nir9696**

### CODE REVIEW

File: app/static/voice.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  let voiceSelect = document.querySelector("#voice");
+
+  
+  function listVoices(){
+    voices = synth.getVoices();
+    let selectedIndex = voiceSelect.selectedIndex < 0 ? 0 : voiceSelect.selectedIndex;
+    
+    voiceSelect.innerHTML = '';
+    for(let voice of voices) {
+      var option = document.createElement('option');
```
Comment (yammesicka) [inline app/static/voice.js:14]: Prefer `let` or `const`

File: app/static/voice.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  let controls = ["volume","pitch","rate"];
+  let actions = ["load","input"];
+  function setOutput(ctrl){
+    let range = document.querySelector(ctrl);
+    range.nextElementSibling.textContent = range.value;
+  }
+  for(let control of controls){
+    for(let action of actions){
+      setOutput(`#${control}`);
+      document.querySelector(`#${control}`).addEventListener(action, (event) => {
```
Comment (yammesicka) [inline app/static/voice.js:41]: `document.getElementById`

File: app/static/voice.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+  }
+  
+  
+  document.querySelector("#speak").addEventListener("click", (event) => {
+    event.preventDefault();
+    let say = document.querySelector("#say").value;
+    let sayThis = new SpeechSynthesisUtterance(say);
+    
+    let vox = voiceSelect.selectedOptions[0].getAttribute('data-name');
```
Comment (yammesicka) [inline app/static/voice.js:53]: Prefer `dataset` instead of `getAttribute`

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/static/voice.js (hunk 1 above)
Comment (nir9696) [reply app/static/voice.js]: thanks :)
Done for this and other declarations

File: app/static/voice.js (hunk 2 above)
Comment (nir9696) [reply app/static/voice.js]: done :)

File: app/static/voice.js (hunk 3 above)
Comment (nir9696) [reply app/static/voice.js]: done :)

**Comment by yammesicka** [comment]
Please make the build work :)

File: app/static/voice.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+window.addEventListener("load",(event) => {
+ 
+  const synth = window.speechSynthesis;
+  let voices = [];
+  const voiceSelect = document.getElementById(voice);
+
+  
+  function listVoices(){
+    voices = synth.getVoices();
+    const selectedIndex = voiceSelect.selectedIndex < 0 ? 0 : voiceSelect.selectedIndex;
```
Comment (yammesicka) [inline app/static/voice.js:10]: Prefer using `Math.max`

File: app/static/voice.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    
+    voiceSelect.innerHTML = '';
```
Comment (yammesicka) [inline app/static/voice.js:12]: Use innerText instead, if possible

File: app/static/voice.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    synth.onvoiceschanged = listVoices;
+  }
+  
+  const controls = ["volume","pitch","rate"];
+  const actions = ["load","input"];
+  function setOutput(ctrl){
+    const range = document.getElementById(ctrl);
+    range.nextElementSibling.textContent = range.value;
+  }
+  for(const control of controls){
```
Comment (yammesicka) [inline app/static/voice.js:38]: Please use ESLint to fix indentation/spacing problems ^^"

File: app/static/voice.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
+window.addEventListener("load",(event) => {
+ 
+  const synth = window.speechSynthesis;
+  let voices = [];
+  const voiceSelect = document.getElementById(voice);
+
+  
+  function listVoices(){
```
Comment (yammesicka) [inline app/static/voice.js:8]: Prefer defining the functions outside the addEventListener

File: requirements.txt [hunk 8]
```diff
@@ -1,103 +0,0 @@
-aiofiles==0.6.0
```
Comment (yammesicka) [inline requirements.txt:1]: Mmm?

**Review by yammesicka — CHANGES_REQUESTED** [review]
Please make the build work :) It's probably the deletion of the content from requirements.txt

File: requirements.txt (hunk 8 above)
Comment (nir9696) [reply requirements.txt]: For some reason the file is truncated and strange things happen to it. I updated again hoping it would be resolved

**Comment by nir9696** [comment]
There is a recurring problem with "datetime" detection in event.py. At first I saw the source of the error - using the "dt" alias, but now there is no such alias, so the error is not clear to me. I saw that others also encountered this error. Can you please take a look?

File: app/templates/event/partials/edit_event_details_tab.html [hunk 9]
```diff
@@ -22,7 +48,10 @@
 </div>
 
 <div class="form_row textarea">
-    <textarea name="description" placeholder="Description"></textarea>
+    <textarea id="say" name="description" placeholder="Description"></textarea>
+  <div>
```
Comment (yammesicka) [inline app/templates/event/partials/edit_event_details_tab.html:52]: Extra empty div here

File: app/templates/event/partials/edit_event_details_tab.html [hunk 10]
```diff
@@ -1,5 +1,31 @@
...
+  <div class="panel">
+    
+    <div class="field">
+      <label for="volume">Volume</label>
+      <input  type="range" id="volume" name="volume" min="0" max="1" step="0.01" value="0.5">
+      <output for="volume"></output>
+    </div>
+    <div class="field">
+      <label for="pitch">Pitch</label>
+      <input type="range" id="pitch" name="pitch" min="0" max="2" step="0.01" value="1"><!-- default 1 -->
```
Comment (yammesicka) [inline app/templates/event/partials/edit_event_details_tab.html:21]: Remove comment

File: app/static/voice.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
+function listVoices(synth, voiceSelect) {
+	voices = synth.getVoices();
```
Comment (yammesicka) [inline app/static/voice.js:2]: Prefer indentation using 2 whitespaces (max 4, but not 8)

Comment (yammesicka) [inline app/static/voice.js:2]: Define the variable: `let voices`/`const voices`

File: app/static/voice.js [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+	const selectedIndex = Math.max(0, voiceSelect.selectedIndex);
+
+	voiceSelect.innerText = '';
+	for (const voice of voices) {
+		const option = document.createElement('option');
```
Comment (yammesicka) [inline app/static/voice.js:7]: Extract to another function

File: app/static/voice.js [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+
+function setOutput(ctrl) {
+	const range = document.getElementById(ctrl);
+	range.nextElementSibling.textContent = range.value;
+}
+
+
+window.addEventListener("load", (event) => {
```
Comment (yammesicka) [inline app/static/voice.js:27]: This function is too complicated - can you please split it into multiple functions?

**Review by yammesicka — CHANGES_REQUESTED** [review]

**Comment by yammesicka** [comment]
> There is a recurring problem with "datetime" detection in event.py. At first I saw the source of the error - using the "dt" alias, but now there is no such alias, so the error is not clear to me. I saw that others also encountered this error. Can you please take a look?

Try to pull from develop, I believe

File: app/static/voice.js (hunk 11 above)
Comment (nir9696) [reply app/static/voice.js]: I don't see 8... but checked it again :)

File: app/static/voice.js [hunk 14]
```diff
@@ -0,0 +1,82 @@
+function listVoices(synth, voiceSelect) {
+	const voices = synth.getVoices();
```
Comment (yammesicka) [inline app/static/voice.js:2]: Be consistent with the amount of spaces in indentations.

Make sure you have no tabs, and 2 spaces for each intendation.

File: app/templates/event/partials/edit_event_details_tab.html [hunk 15]
```diff
@@ -1,53 +1,66 @@
...
+      <input type="range" id="rate" name="rate" min="0.1" max="10" step="0.1" value="1">
+      <output for="rate"></output>
+   </div>
 </div>
-
 <div class="form_row">
-  <input type="date" name="end_date" placeholder="End Date" required>
-  <input type="time" name="end_time" placeholder="End Time" required>
+   <input type="date" name="start_date" placeholder="Start Date" required>
+   <input type="time" name="start_time" placeholder="Start Time" required>
```
Comment (yammesicka) [inline app/templates/event/partials/edit_event_details_tab.html:31]: Please add labels for a11y

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/static/voice.js (hunk 14 above)
Comment (nir9696) [reply app/static/voice.js]: I hope now the indentations are correct :)

File: app/templates/event/partials/edit_event_details_tab.html [hunk 16]
```diff
@@ -1,53 +1,74 @@
+<script src="{{ url_for('static', path='/voice.js') }}"></script>
```
Comment (yammesicka) [inline app/templates/event/partials/edit_event_details_tab.html:1]: Move this to be before the `</body>`

File: app/templates/event/partials/edit_event_details_tab.html [hunk 17]
```diff
@@ -1,53 +1,74 @@
...
+         <option value="blue">Blue</option>
+      </select>
+   </div>
+   <div class="form_row_end">
+	  <label for="availability">Availability:</label>
+      <select id="availability" name="availability">
+         <option value="free">Free</option>
+         <option value="busy" selected>Busy</option>
+      </select>
+	  <label for="privacy">Privacy:</label>
```
Comment (yammesicka) [inline app/templates/event/partials/edit_event_details_tab.html:68]: Unindent

**Review by yammesicka — COMMENTED** [review]
Almost done :)

File: app/templates/event/partials/edit_event_details_tab.html (hunk 17 above)
Comment (nir9696) [reply app/templates/event/partials/edit_event_details_tab.html]: fixed to 3-spaces indentation

File: app/templates/event/eventedit.html [hunk 18]
```diff
@@ -1,40 +1,41 @@
...
-        </div>
-    </form>
-{% endblock content %}
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Edit Event</title>
+  <link href="{{ url_for('static', path='event/eventedit.css') }}" rel="stylesheet">
+  <script src="{{ url_for('static', path='/voice.js') }}"></script>
```
Comment (yammesicka) [inline app/templates/event/eventedit.html:7]: Move this line to be before the `</body>`

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/templates/event/eventedit.html (hunk 18 above)
Comment (nir9696) [reply app/templates/event/eventedit.html:7]: I hope now that's OK :)

File: app/templates/event/eventedit.html [hunk 19]
```diff
@@ -1,40 +1,41 @@
...
-    </form>
-{% endblock content %}
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Edit Event</title>
+  <link href="{{ url_for('static', path='event/eventedit.css') }}" rel="stylesheet">
+</head>
+<script src="{{ url_for('static', path='/voice.js') }}"></script>
```
Comment (yammesicka) [inline app/templates/event/eventedit.html:8]: Move to one line before the `</body>`

Comment (nir9696) [reply app/templates/event/eventedit.html]: sorry for the misunderstanding... fixed :)

**Review by yammesicka — APPROVED** [review]

**Comment by OdeChan** [comment]
Hi, some hindsight notes: notice that in edit_event_details_tab.html you are using multiple **id** tags with the same values that is bad practice (there should be only one element with the same id tag per page) + you have div elements without closing tags which breaks the layout.

**Pull request merged**